bash_completion

```diff
@@ -895,7 +895,11 @@ _comp_delimited()
         _comp_compgen COMPREPLY "$@" -- "${cur##*"$delimiter"}"
     fi
 
-    ((${#COMPREPLY[@]} == 1)) && COMPREPLY=("$prefix$COMPREPLY")
+    local i
+    for i in "${!COMPREPLY[@]}"; do
+        COMPREPLY[i]="$prefix${COMPREPLY[i]}"
+    done
```

Comment:
While this certainly seems to work, it's also somewhat unfortunate, because displaying the completions with the prefix in place makes them harder to read, and with `show-all-if-ambiguous` on it appears we wouldn't have to do it (I'm biased, I have it on). So I'm contemplating something like:

```suggestion
    # If `show-all-if-ambiguous` is off, existing completions preceding the
    # comma and the comma itself are lost if there are multiple completions
    # offered for the thing after the prefix. This does not seem to happen
    # with it on, so avoid repeating the prefix in that case for readability.
    # https://github.com/scop/bash-completion/pull/532#issuecomment-873675676
    if _comp_readline_variable_on show-all-if-ambiguous; then
        ((${#COMPREPLY[@]} == 1)) && COMPREPLY=("$prefix$COMPREPLY")
    else
        local i
        for i in "${!COMPREPLY[@]}"; do
            COMPREPLY[i]="$prefix${COMPREPLY[i]}"
        done
    fi
```

I suppose this would require some addressing/complexity in tests, too.

Thoughts?

Reply:
As we have discussed in the past in #546, the current test framework totally relies on `set show-all-if-ambiguous on`, so I'm afraid there is no way to test the behavior for the case `set show-all-if-ambiguous off`.

Reply:
@akinomyoga showed that `((${#COMPREPLY[@]} == 1)) && COMPREPLY=("$prefix$COMPREPLY")` isn't guaranteed to work even if `show-all-if-unmodified` is `on` (https://github.com/scop/bash-completion/pull/913#issuecomment-1490140309). Copying the snippet here:
```
$ bind 'set show-all-if-unmodified on'
$ echo x,y,he[TAB]
hello1  hello2  hello3
$ echo x,y,he         # The original word is preserved
$ echo x,y,h[TAB]
hello1  hello2  hello3
$ echo hello          # The original word is lost
```

I'm not an expert here, but it feels very brittle to be relying on some obscure undocumented interaction between options, that might break easily down the road (as showcased above).

Reply:
Ah, yes. I was also about to write it. I tried the change suggested by @scop with `set show-all-if-ambiguous` and confirmed that it still fails. Also, I was checking if it is possible to detect the insertion by looking at `COMP_TYPE`, but there doesn't seem to be a reliable way.

Reply:
Ok. Then let's just go with the implementation here that is known to work in all cases, you guys have spent more than enough time here already. Thanks!

Reply:
Added a note in e3840d0f9e5ed0d9c16bfb4cd1d9d63198292886 as a reminder to take stuff into account if someone is tempted to revisit this later.